jdbc-perf-logger-driver/src/main/java/ch/sla/jdbcperflogger/logger/PerfLoggerClientThread.java

```diff
@@ -83,6 +85,17 @@ public void run() {
                 LOGGER.info("Error in connection with " + socketAddress + ", will try again later", e);
             }
         }
+        LOGGER.debug("Client for " + socketAddress + "closed");
+    }
+
+    @Override
+    public void close() {
+        try {
+            socket.close();
+        } catch (final IOException e) {
+            LOGGER.info("Error closing socket at port" + socket.getLocalPort());
+        }
+        done = true;
```

Comment:
you'd better set done to true before closing the socket. This will avoid a race condition in run() where the loop over while(!done) could run once more (even if unprobable).